features/net/network-socket/NetworkStack.cpp

```diff
@@ -20,11 +20,29 @@
 #include "stddef.h"
 #include <new>
 
+static bool stackHasIPV4(NetworkStack *stack)
+{
+    const char *ip = stack->get_ip_address();
+    return ip && strchr(ip, '.');
+}
 
 // Default NetworkStack operations
 int NetworkStack::gethostbyname(SocketAddress *address, const char *name)
 {
-    return nsapi_dns_query(this, address, name);
+    nsapi_version_t version;
+
+    // First check for and handle literal addresses
+    if (address->set_ip_address(name)) {
+        return 0;
+    }
+
+    if (stackHasIPV4(this)) {
```

Comment:
Doesn't make sense to me. Surely nsapi_dns_query should do what it name says - a DNS query. Having to add literal parsing to the DNS code too doesn't make sense - it has to be outside the DNS code anyway.

It's gethostbyname that has to handle literals before attemping a DNS query.